src/models/stores/serial.ts

```diff
@@ -100,7 +92,43 @@ export class SerialDevice {
     }
   }
 
-  public handleStreamObj(value: string, channels: Array<NodeChannelInfo>){
+  public handleMicroBitStreamObj(value: string, channels: Array<NodeChannelInfo>){
+    this.localBuffer += value;
+
+    // [sc]   signal or control
+    // [abcd] which micro:bit
+    // [rth]  relay, temp, humidity
+    const pattern = /([sc]{1})([abcd]{1})([rth]{1})([0-9.]+)\s{0,}[\r][\n]/g;
+    let match: RegExpExecArray | null;
+
+    do {
+      match = pattern.exec(this.localBuffer);
+      if (!match) break;
+
+      const [fullMatch, signalType, microbitId, element, reading] = match;
+      this.localBuffer = this.localBuffer.substring(match.index + fullMatch.length);
+
+      const targetChannelId = `${element}-${microbitId}`;
```

Comment:
In the future I think you'll want each relay to result to go to its own channel instead of a single channel for all of them. But it doesn't make sense to change that now.  Probably, you should wait until you have an implementation of the relay output block commands. It is those  relay output blocks where you'll want to use this relay input data, so the user can see the actual status of the relay instead of just what they told it to be.  So those output blocks need to monitor the the relay values that are coming from the microbit.

Reply:
In the next PR there is a more relay-specific channel setup for the output nodes to "tune in" to, but I think I was still thinking one "relays channel" per microbit hub as we had talked about sending the state of each relay back with each change.  E.g. `sar001`, but I could still establish a dedicated channel per relay.  Anyway I'll try that and flag what ends up happening in the next PR.